suffix_update.py

```diff
@@ -0,0 +1,132 @@
+import sys
+is_py_2 = sys.version_info[0] == 2
+
+
+def read_suffix_list(content):
+    '''
+    Read and parse suffix list.
+    '''
+    # remove comment & blank line
+    suffix_list = [line.lower().strip()
+                   for line in contents
+                   if not line.startswith('//') and line.strip()]
+
+    # find idna decodable suffixs
+    idna_encoded_suffix_list = []
+    for suffix in suffix_list:
+        try:
+            encoded = suffix.encode('idna').decode('ascii')
+            if suffix != encoded:
+                idna_encoded_suffix_list.append(encoded)
+        except Exception:
+            pass
+
+    idna_encoded_suffix_list.sort()
+    # extend idna encoded suffix_list
+    suffix_list.extend(idna_encoded_suffix_list)
+
+    suffix_list.sort()
+
+    return suffix_list
+
+
+def write_suffix_py(o, suffix_list):
+    '''
+    Write suffix list as python source.
+    '''
+    o.write("# -*- coding: utf-8 -*-\n"
+            "from __future__ import unicode_literals\n\n")
+
+    def write(var_begin, suffix_lst, end, line_acc, process, process_elem):
+        first_line_max_length = 79 - len(var_begin)
+        line_max_length = first_line_max_length
+        one_line_max_length = first_line_max_length
+        count, tmp, indent = 0, [], 0
+        o.write(var_begin)
+        for suffix in suffix_lst:
+            processed_elem, length = process_elem(suffix)
+            length += len(processed_elem)
+            if count + len(processed_elem) + line_acc >= line_max_length:
+                o.write((' ' * indent + process(tmp, indent == 0)))
+                o.write('\n')
+                tmp, count, indent = [], 0, len(var_begin)
+                line_max_length = one_line_max_length
+            tmp.append(processed_elem)
+            count += length
+        if tmp:
+            o.write(' ' * indent + process(tmp, indent == 0))
+        o.write(end)
+
+    # Write suffixes list.
+    #
+    # write('SUFFIXES = [', suffix_list, ' ]', 1,
+    #       lambda x, f: ', '.join(x) + ',',
+    #       lambda x: ('"%s"' % x, 2))
+
+    # o.write('\n\n')
+
+    # write public suffixes Regular Expression.
+    suffix_list = list(map(lambda x: x.replace('.', '\\.')
+                           .replace('*', '[^.]+').replace('!', ''),
+                           suffix_list))
+    suffix_list.sort()
+    suffix_list.reverse()
+    write('SUFFIXES_RE = (', suffix_list, ')', 3,
+          lambda x, f: ('"%s"' if f else '"|%s"') % '|'.join(x),
+          lambda x: (x, 1))
+    o.write('\n')
```

Comment:
Seems like this function is doing more than "write suffix list as python source", but it's pretty complicated and very little of it is commented, so I don't really understand what it's doing and why.

Can you add more explanatory comments as well as a more detailed docstring covering what the output should look like?